botocore/utils.py

```diff
@@ -925,12 +925,33 @@ def set_request_url(self, params, context, **kwargs):
         if endpoint is not None:
             params['url'] = _get_new_endpoint(params['url'], endpoint)
 
-    def redirect_from_cache(self, params, context, **kwargs):
+    def modify_request_before_send(self, params, context, model, **kwargs):
+        bucket = params.get('Bucket')
+        self.redirect_from_cache(bucket, context)
+        if model.name == 'CreateBucket':
+            self.redirect_create_bucket(bucket, params, context)
+
+    def redirect_create_bucket(self, bucket, params, context):
+        configuration = params.get('CreateBucketConfiguration', {})
+        location = configuration.get('LocationConstraint', None)
+
+        if location is not None:
```

Comment:
So do we know for certain that only ``us-east-1`` is the only region that allows for an empty ``LocationConstraint``? With the way you have it now, the default endpoint for ``aws`` will always be used. And so if you are in a different partition that had a default endpoint, it would not use that.